lib/Scheduler/sched_basic_data.cpp

```diff
@@ -281,23 +288,27 @@ bool SchedInstruction::ApplyPreFxng(LinkedList<SchedInstruction> *tightndLst,
 
 void SchedInstruction::AddDef(Register *reg) {
   if (defCnt_ >= MAX_DEFS_PER_INSTR) {
-    llvm::report_fatal_error("An instruction can't have more than " +
-                                 std::to_string(MAX_DEFS_PER_INSTR) + " defs",
-                             false);
+    llvm::report_fatal_error(
+        llvm::StringRef("An instruction can't have more than " +
+                        std::to_string(MAX_DEFS_PER_INSTR) + " defs"),
+        false);
   }
   // Logger::Info("Inst %d defines reg %d of type %d and physNum %d and useCnt
   // %d",
   // num_, reg->GetNum(), reg->GetType(), reg->GetPhysicalNumber(),
   // reg->GetUseCnt());
   assert(reg != NULL);
+  // if (IsRoot()) Logger::Info("added %d to defs of root", reg->GetNum());
   defs_[defCnt_++] = reg;
 }
 
 void SchedInstruction::AddUse(Register *reg) {
   if (useCnt_ >= MAX_USES_PER_INSTR) {
-    llvm::report_fatal_error("An instruction can't have more than " +
-                                 std::to_string(MAX_USES_PER_INSTR) + " uses",
-                             false);
+    printMF();
```

Comment:
Why are we unconditionally printing MF? Should printMF() be changed to only print under certain `#ifdef`s?

Reply:
Yeah this needs to be handled